src/js/angular/core/directives/queryeditor/query-editor.controller.js

```diff
@@ -539,14 +539,14 @@ function QueryEditorCtrl($scope, $timeout, toastr, $repositories, $modal, ModalS
                 toastr.success($translate.instant('query.editor.save.saved.query.success.msg', {name: query.name}));
             })
             .error(function (data) {
-                let msg = getError(data);
+                const msg = getError(data);
                 toastr.error(msg, $translate.instant('query.editor.create.saved.query.error'));
-                // TODO: This condition will always be true
-                if (msg = "Query '" + query.name + "' already exists!") {
-                    query.query = query.body;
-                    const queryExists = true;
-                    $scope.saveQuery(query, queryExists);
+                let queryExists = false;
+                if (msg === "Query '" + query.name + "' already exists!") {
```

Comment:
This check will fail if the interface is in french for example. Can't we check only for the query name instead of the whole message?!

Reply:
This message comes from the backend and it is always in English. The one used in the modal is different (query.sample.existing.query.warning in locale-en or locale-fr).

Reply:
Currently there are no other errors containing the query name that can be returned from the controller, so a check only on the query name would be valid. 
But currently if the query exists, the controller returns status 500 with the error message, which I believe should be fixed?

Reply:
We don't have a practise until now to return the error message codes in public APIs and we cannot change this in minor versions. The fact that the status code is 500 is a bug, and can be fixed for 10.1, so let's make the status code 409 and from there the workbench can show whatever message we want to. So a simple check of the status code here would work.